core/src/main/java/org/elasticsearch/index/query/BaseTermQueryBuilder.java

```diff
@@ -178,10 +178,10 @@ protected void doXContent(XContentBuilder builder, Params params) throws IOExcep
     public QueryValidationException validate() {
         QueryValidationException validationException = null;
         if (fieldName == null || fieldName.isEmpty()) {
-            validationException = QueryValidationException.addValidationError("field name cannot be null or empty.", validationException);
+            validationException = QueryValidationException.addValidationError(getName(), "field name cannot be null or empty.", validationException);
```

Comment:
I see how this gives context to the error message, wondering if there is an easy way to be more specific in case there is more than query with the same name in the complete query (like a `match` in both must and should clause of BooleanQueryBuilder). I guess that's not possible right now, but would be great if inner queries knew about their parents, then we could output the full path here.